log/experimental_level/level_test.go

```diff
@@ -148,10 +149,41 @@ func TestContextLevel(t *testing.T) {
 	var logger log.Logger
 	logger = log.NewLogfmtLogger(&buf)
 	logger = log.NewContext(logger).With("caller", log.Caller(5))
-	logger = level.New(logger, level.Allowed(level.AllowAll()))
+	logger = level.NewFilter(logger, level.AllowAll())
 
 	level.Info(logger).Log("foo", "bar")
-	if want, have := `caller=level_test.go:153 level=info foo=bar`, strings.TrimSpace(buf.String()); want != have {
-		t.Errorf("want %q, have %q", want, have)
+	if want, have := `caller=level_test.go:154 level=info foo=bar`, strings.TrimSpace(buf.String()); want != have {
+		t.Errorf("\nwant '%s'\nhave '%s'", want, have)
+	}
+}
+
+func TestLevelFormatting(t *testing.T) {
+	testCases := []struct {
+		name   string
+		format func(io.Writer) log.Logger
+		output string
+	}{
+		{
+			name:   "logfmt",
+			format: log.NewLogfmtLogger,
+			output: `level=info foo=bar`,
+		},
+		{
+			name:   "JSON",
+			format: log.NewJSONLogger,
+			output: `{"foo":"bar","level":"info"}`,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			var buf bytes.Buffer
+
+			logger := tc.format(&buf)
+			level.Info(logger).Log("foo", "bar")
+			if want, have := tc.output, strings.TrimSpace(buf.String()); want != have {
+				t.Errorf("\nwant: '%s'\nhave '%s'", want, have)
```

Comment:
`%q` works the same on all platforms as it is geared towards escaping strings like you would see in Go source code. It is often unsatisfactory when printing Windows file paths because the `\`s get doubled up, but that's not my motivation here. In this case the motivation is that I added a test that outputs JSON and didn't want to see lots of `\"` in the output.

IMO, `%q` should be used when the intent is for escaped strings. If you just want to demarcate the beginning and ending of a string (to spot extra white space or spot empty strings) then explicit markers are preferred.

Of course now I am thinking that perhaps `` ` `` would be better than `'` for Go because of its use within the language for unescaped string literals.